### PR TITLE
Bundle fixture classes

### PR DESCRIPTION
**Pull request by twojtylak**

### CODE REVIEW

**Review by michz — DISMISSED** [review]

**Review by michz — DISMISSED** [review]

File: src/Fixture/Factory/ProductBundleExampleFactory.php [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+    private function createSlotOptions(array $rawSlotOptions = []): ProductBundleSlotOptionsInterface
+    {
+        $slotOptions = new ProductBundleSlotOptions();
+        foreach ($rawSlotOptions as $optionName => $optionValue) {
+            $setter = 'set' . ucfirst($optionName);
+
+            Assert::methodExists($slotOptions, $setter, sprintf('Setter %s for ProductBundleSlotOptions is not defined', $setter));
+
+            if (method_exists($slotOptions, $setter = 'set' . ucfirst($optionName))) {
+                $slotOptions->$setter($optionValue);
```
Comment (solutionDrive-Alt) [inline src/Fixture/Factory/ProductBundleExampleFactory.php]: Just as a note, there is also the possibility to add additional options with the 
[ProductBundleSlotOptionsInterface::addOptionMethod](https://github.com/solutionDrive/SyliusProductBundlesPlugin/blob/84d57030be3646700c18a98d4c3d1df413625c48/src/Service/Options/OptionsInterface.php#L14) in an else-branch if that is a thing that is needed.

Comment (mattagohni) [inline src/Fixture/Factory/ProductBundleExampleFactory.php:105]: There is also the possibility to add additional options with the [ProductBundleSlotOptionsInterface::addOptionMethod](https://github.com/solutionDrive/SyliusProductBundlesPlugin/blob/84d57030be3646700c18a98d4c3d1df413625c48/src/Service/Options/OptionsInterface.php#L14) if there is a use case for this. But maybe this is more important in the project, that uses this plugin

File: src/Resources/config/services.yml [hunk 2]
```diff
@@ -19,3 +19,17 @@ services:
         arguments:
             - "@solutiondrive.factory.product_bundle"
             - "@solutiondrive.factory.product_bundle_slot"
+
+    solutiondrive.product_bundles.fixture.example_factory.product_bundle:
```
Comment (mattagohni) [inline src/Resources/config/services.yml:20]: Should the example-factories be defined in the 'main'-services.yml? Maybe you don't really want these services in your prod-environment...

File: src/Fixture/Factory/ProductBundleExampleFactory.php (hunk 1 above)
Comment (twojtylak) [reply src/Fixture/Factory/ProductBundleExampleFactory.php:105]: yeah thats better. I'll change this tomorrow. No idea why I didn't see this function.

File: src/Resources/config/services.yml (hunk 2 above)
Comment (twojtylak) [reply src/Resources/config/services.yml:20]: Neither sylius nor any plugin I've seen makes a difference between prod and dev-services. So I don't think we should..

File: src/Fixture/Factory/ProductBundleExampleFactory.php (hunk 1 above)
Comment (twojtylak) [reply src/Fixture/Factory/ProductBundleExampleFactory.php:105]: Hmm I think for now we don't need additional options. There are not used at the moment anyway. Maybe the naming is a little bit confusing (they are saved in a variable additionalOptions) ...

File: src/Resources/config/services.yml (hunk 2 above)
Comment (mattagohni) [reply src/Resources/config/services.yml:20]: ok, was only a thought... :-)

**Comment by mattagohni** [comment]
@twojtylak  can you please rebase? we have conflicts.

File: src/Fixture/Factory/ProductBundleExampleFactory.php (hunk 1 above)
Comment (mattagohni) [reply src/Fixture/Factory/ProductBundleExampleFactory.php:105]: @twojtylak yeah, that's correct. The idea behind that is, that you have fixed options for a ProductBundleSlotOptions-Object (position/isPresentationSlot) and the possibility to add custom options that can influence the BundleSlotCreation. (e.g. when someone implements his own BundleSlot-Class).

**Comment by twojtylak** [comment]
I rebased the branch right now. There were some  integrity constraint violations when you tried to delete a ProductBundle with assigned Slots. So I added the onDelete cascade

**Review by mattagohni — APPROVED** [review]

**Pull request merged**